Search history: Design review

ashiagr opened the pull request:
Part of #752

## Description

This includes search history design improvements to match the specs: 8JZyr1ZkWiGn3kSjJYvpoZ-fi-160%3A5856
APK for design review attached in pdeCcb-1iH-p2#comment-1675.

## Screenshots or Screencast 

Light Theme | Different Theme | Large font
---|----|---
<img width=320 src="https://user-images.githubusercontent.com/1405144/218469631-66d404a4-d0be-482b-9cfb-aeb9e3567091.png"/> | <img width=320 src="https://user-images.githubusercontent.com/1405144/218470639-0dc77c20-eca4-4bea-8d2b-88ef3c7a172c.png"/>| <img width=320 src="https://user-images.githubusercontent.com/1405144/218472178-089026aa-d29c-4e3a-abd7-efd3a115bb7d.png"/>


|Landscape mode|
|----|
|<img width=700 src="https://user-images.githubusercontent.com/1405144/218472003-2cc10c62-b204-4d97-9083-a0e2a9f794d5.png"/>|



## Checklist
- If this is a user-facing change, I have added an entry in CHANGELOG.md
- I have considered whether it makes sense to add tests for my changes
- [x] All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`
- [x] Any jetpack compose components I added or changed are covered by compose previews
 
#### I have tested any UI changes...
<!-- If this PR does not contain UI changes, ignore these items -->
- [x] with different themes
- [x] with a landscape orientation
- [x] with the device set to have a large display and font size
- [x] for accessibility with TalkBack

CookieyedCodes commented:
Would it be a nice improvement that if a searched podcast is not subscribed that their is a plus icon to subscribe to it on from the history screen(episodes & folders would have a go to folder & go to episode icon representing by an arrow instead > 🤔)

ashiagr commented:
> Would it be a nice improvement that if a searched podcast is not subscribed that their is a plus icon to subscribe to it on from the history screen(episodes & folders would have a go to folder & go to episode icon representing by an arrow instead > 🤔)

We're eliminating subscribe icon from search history rows as it felt too cluttered to have it along with the close button and to keep search history rows lightweight. It will be present in the search result rows. With that, there's no need to show plus or arrow icon.

> Also maby have a setting for a condensed view the padding feels a tad excessive 🤔🤔

Nice idea :) thanks for sharing!

CookieyedCodes commented:
That makes more sense 😅, pinning if anything would still be nice at that tho 🤔 (maby on a long press if you want to cut down on icons 🤔🤔)

ashiagr commented:
Addressed feedback: pdeCcb-1iH-p2#comment-1693 
cc @adamzelinski 

> Can we please update the search field empty string to say ‘Search podcasts or add URL’

> Can we please update the delete button to read ‘Clear Search History’ I think delete seems a bit harsh.

Search field empty text | Clear Search History 
---|---
<img width=300 src="https://user-images.githubusercontent.com/1405144/219335363-48b492b4-2494-43cf-a889-8ce2428fb20b.png"/> | <img width=300 src="https://user-images.githubusercontent.com/1405144/219335346-22bc11e8-a9bd-4c50-8bd2-424c31492b7a.png"/>

CookieyedCodes commented:
I'd personally say add RSS URL that would give better clarity in my view 🤔🤔

mchowning reviewed (approved):
This looks great. Nice job! 🎉 

The only minor issue I noticed was that when tapping on a search term in the history, it looks like we still have the couple of seconds delay before the remote search is initiated (like if you had typed the search term). I wonder if it might be better to instantly do start the remote search in that scenario. It's most noticeable when searching from the Discover screen. 

Certainly not a blocker for merging this PR, just something to think about going forward.

<details>
<summary>Video</summary>

https://user-images.githubusercontent.com/4656348/219454841-7e0eb0aa-3976-4467-a6ba-bf09ab909d1c.mov


</details>

ashiagr commented on modules/services/model/src/main/java/au/com/shiftyjelly/pocketcasts/models/db/dao/SearchHistoryDao.kt:
```diff
@@ -25,4 +24,7 @@ abstract class SearchHistoryDao {
     @Query("DELETE FROM search_history")
     abstract suspend fun deleteAll()
+
+    @Query("DELETE FROM search_history where _id NOT IN (SELECT _id from search_history ORDER BY modified DESC LIMIT :limit)")
```
Thank you for adding the test! It is definitely useful. 🙏

ashiagr commented:
> The only minor issue I noticed was that when tapping on a search term in the history, it looks like we still have the couple of seconds delay before the remote search is initiated (like if you had typed the search term). I wonder if it might be better to instantly do start the remote search in that scenario. It's most noticeable when searching from the Discover screen.

Good point. I fixed it in 4227b91.

adamzelinski commented:
> I'd personally say add RSS URL that would give better clarity in my view

You're right, it might be better to use add RSS URL. Can we please make this change?

ashiagr commented:
> I'd personally say add RSS URL that would give better clarity in my view

> You're right, it might be better to use add RSS URL. Can we please make this change?

Changed in 29956f9.

Thanks everyone for great suggestions 🙏

CookieyedCodes commented:
Well it looks good here 
![Screenshot_2023-02-24-15-53-32-15_28ad70af3b47247953fcd94176b9a9c1.jpg](https://user-images.githubusercontent.com/20846898/221224666-d0d75e21-5a85-4f60-a60c-61af60d6806e.jpg)

But it seems folders share the same text now 
![Screenshot_2023-02-24-15-48-47-56_28ad70af3b47247953fcd94176b9a9c1.jpg](https://user-images.githubusercontent.com/20846898/221224782-92d38c03-f855-4e4b-bc9a-8864d2ce875e.jpg)

ashiagr commented:
Raised beta fix: https://github.com/Automattic/pocket-casts-android/pull/803. Thanks for pointing it out, @CookieyedCodes 🙏
